android/src/main/java/com/internxt/mobilesdk/MobileSdkModule.kt

```diff
@@ -116,6 +115,60 @@ class MobileSdkModule(reactContext: ReactApplicationContext) :
     }
   }
 
+  @ReactMethod
+  fun processPhotosItem(config: ReadableMap, promise: Promise) {
+    try {
+      val plainFilePath  = config.getString("plainFilePath") ?: throw InvalidArgumentException("Missing plainFilePath")
+      val mnemonic = config.getString("mnemonic") ?: throw InvalidArgumentException("Missing mnemonic")
+      val bucketId = config.getString("bucketId") ?: throw InvalidArgumentException("Missing bucketId")
+      val photosUserId = config.getString("photosUserId") ?: throw InvalidArgumentException("Missing photosUserId")
+      val deviceId = config.getString("deviceId") ?: throw InvalidArgumentException("Missing deviceId")
+      Logger.info("Starting to process Photo, creating Thread")
+      val runnable = Runnable {
+        try {
+          Logger.info("Begin of Photo process")
+          val processStart = Performance.measureTime()
+          val outputDir: File = reactApplicationContext.cacheDir
+          val previewDestination = File.createTempFile("tmp", ".jpg", outputDir)
+          val uri = FS.getFileUri(plainFilePath, true)
+          val originalSourceStream = reactApplicationContext.contentResolver.openInputStream(uri) ?: throw IOException("Unable to open preview input stream at ${uri.path}")
+          val previewSourceStream = reactApplicationContext.contentResolver.openInputStream(uri) ?: throw IOException("Unable to open input stream at ${uri.path}")
+
+          // Write here the encrypted photo result
+          val encryptedOriginalDestination = File.createTempFile("tmp_original", ".enc", outputDir)
```

Comment:
Now that we have everything on native, should we still do this? Or is it for caching purposes?

Reply:
We need to store this somewhere in order to upload it, we could use streams but with streams some parts are tricky, so a tmp file is a nice interface since we only write and read once, once all the process is done, those files are removed.

Note that the Java createTempFile method adds an UUID to the file, so the file will be tmp_original1412412412412.enc or something like that, is not the same file all the time

Reply:
Okay, let's leave it for now as is.